cpp/src/io/parquet/experimental/hybrid_scan_helpers.hpp

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "io/parquet/parquet_gpu.hpp"
+#include "io/parquet/reader_impl_helpers.hpp"
+
+#include <cudf/io/detail/utils.hpp>
+#include <cudf/io/experimental/hybrid_scan.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <memory>
+#include <optional>
+#include <vector>
+
+namespace cudf::experimental::io::parquet::detail {
+
+using aggregate_reader_metadata_base = cudf::io::parquet::detail::aggregate_reader_metadata;
+using equality_literals_collector    = cudf::io::parquet::detail::equality_literals_collector;
+using FileMetadata                   = cudf::io::parquet::FileMetaData;
+using inline_column_buffer           = cudf::io::detail::inline_column_buffer;
+using input_column_info              = cudf::io::parquet::detail::input_column_info;
+using metadata_base                  = cudf::io::parquet::detail::metadata;
+using row_group_info                 = cudf::io::parquet::detail::row_group_info;
+
+/**
+ * @brief Class for parsing dataset metadata
+ */
+struct metadata : private metadata_base {
```

Comment:
Derived from `metadata` in `reader_impl_helpers.hpp` allowing file metadata construction from footer bytes